src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -132,19 +134,82 @@ private static Type GetTypeFromManifestName(Stream stream, DeserializerSession s
             });
         }
 
+        public static bool disallowUnsafeTypes = true;
+
+        private static ReadOnlyCollection<string> unsafeTypesDenySet =
+            new ReadOnlyCollection<string>(new[]
+            {
+                "System.Security.Claims.ClaimsIdentity",
+                "System.Windows.Forms.AxHost.State",
+                "System.Windows.Data.ObjectDataProvider",
+                "System.Management.Automation.PSObject",
+                "System.Web.Security.RolePrincipal",
+                "System.IdentityModel.Tokens.SessionSecurityToken",
+                "SessionViewStateHistoryItem",
+                "TextFormattingRunProperties",
+                "ToolboxItemContainer",
+                "System.Security.Principal.WindowsClaimsIdentity",
+                "System.Security.Principal.WindowsIdentity",
+                "System.Security.Principal.WindowsPrincipal",
+                "System.CodeDom.Compiler.TempFileCollection",
+                "System.IO.FileSystemInfo",
+                "System.Activities.Presentation.WorkflowDesigner",
+                "System.Windows.ResourceDictionary",
+                "System.Windows.Forms.BindingSource",
+                "Microsoft.Exchange.Management.SystemManager.WinForms.ExchangeSettingsProvider",
+                "System.Diagnostics.Process",
+                "System.Management.IWbemClassObjectFreeThreaded"
+            });
+
+#if NETSTANDARD1_6
+#else
+        public static bool UnsafeInheritanceCheck(Type type)
+        {
+            if (type.IsValueType)
+                return false;
+            var currentBase = type.BaseType;
```

Comment:
should we actually start this off with type, just to be safe?

Reply:
In terms of security scanning?

Reply:
If that's your question, then yes I would - want to work our way from bottom to top

Reply:
No, its fine, its the while...loop check that needs to be changed

Reply:
Ah ok, I misunderstood what was being asked here